src/Middleware/OutputCaching/src/OutputCacheContext.cs

```diff
@@ -42,17 +37,17 @@ internal OutputCacheContext(HttpContext httpContext, IOutputCacheStore store, Ou
     /// <summary>
     /// Gets the <see cref="HttpContext"/>.
     /// </summary>
-    public HttpContext HttpContext { get; }
+    public required HttpContext HttpContext { get; init; }
 
     /// <summary>
-    /// Gets the response time.
+    /// Gets or sets the response time.
     /// </summary>
-    public DateTimeOffset? ResponseTime { get; internal set; }
+    public DateTimeOffset? ResponseTime { get; set; }
 
     /// <summary>
     /// Gets the <see cref="CacheVaryByRules"/> instance.
     /// </summary>
-    public CacheVaryByRules CacheVaryByRules { get; set; } = new();
+    public CacheVaryByRules CacheVaryByRules { get; internal set; } = new();
```

Comment:
internal set?

Reply:
It's set only in the middleware. Note that it could also be private and mutated directly I think. Is `internal` an issue in this case? It doesn't need to be public.

Reply:
This came up in API review, and since `CacheVaryByRules` it's pre-initialized to and empty value and is fully mutable, we decided that get-only is okay.

But why does this ever need to be set? Can't we just set the `CacheVaryByRules` sub-properties in the middleware? Not a huge deal, bhat way if someone captures the `CacheVaryByRules` instance in their `IOutputCachePolicy.CacheRequestAsync()` instance before the middleware populates the rules in `TryServeFromCacheAsync`/`CreateCacheKey`, the reference will still be valid.